pre-commit.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+"""
+Pre-commit hook that checks errors with pyflakes.
+Adapted from: https://github.com/cbrueffer/pep8-git-hook
+"""
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+
+
+def system(*args, **kwargs):
+    """
+    Execute a system process and returns the output, errors and exit code
+    """
+    kwargs.setdefault('stdout', subprocess.PIPE)
+    proc = subprocess.Popen(args, **kwargs)
+    out, err = proc.communicate()
+    return out, err, proc.returncode
+
+
+def find_modified_and_added_py_files():
+    """
+    Find staged python files that are either modified or added files. This
+    method ignores deleted files.
+    """
+    modified = re.compile('^[AM]+\s+(?P<name>.*\.py$)', re.MULTILINE)
+    files, err, code = system('git', 'status', '--porcelain')
+
+    if code != 0:
+        raise Exception(err)
+
+    return modified.findall(files.decode("utf-8"))
+
+
+def create_temp_dir_staged(files):
```

Comment:
This is nice but I think we should try another way. This copy the diff from the files in a temp area, which is nice, but the lines which the errors appear is different than the original file, which is not nice.

Any ideia @CheesecakeLabs/backend?